tuf_conformance/test_repository_simulator.py

```diff
@@ -0,0 +1,81 @@
+import unittest
+from tempfile import TemporaryDirectory
+
+from tuf.api.metadata import (
+    Metadata,
+    Root,
+    Snapshot,
+    Targets,
+    Timestamp,
+)
+
+from tuf_conformance import utils
+from tuf_conformance.simulator_server import SimulatorServer
+
+
+class TestRepositorySimulator(unittest.TestCase):
+    def test_repo_initialzation_unittest(self) -> None:
+        tmp_dir = TemporaryDirectory()
+        server = SimulatorServer(dump_dir=tmp_dir.name)
+        init_data, repo = server.new_test("test_repo_initialzation")
+        self.assertEqual(repo.root.version, 1)
+        self.assertEqual(repo.timestamp.version, 1)
+        self.assertEqual(repo.snapshot.version, 1)
+        self.assertEqual(repo.targets.version, 1)
+        self.assertEqual(len(repo.signed_mds), 4)
+        self.assertEqual(
+            Metadata.from_bytes(repo.signed_mds[Root.type][-1]).signed.version, 1
+        )
+        self.assertEqual(
+            Metadata.from_bytes(repo.signed_mds[Timestamp.type][-1]).signed.version, 1
+        )
+        self.assertEqual(
+            Metadata.from_bytes(repo.signed_mds[Snapshot.type][-1]).signed.version, 1
+        )
+        self.assertEqual(
+            Metadata.from_bytes(repo.signed_mds[Targets.type][-1]).signed.version, 1
+        )
+
+    def test_basic_metadata_hash_support_unittest(self) -> None:
+        tmp_dir = TemporaryDirectory()
+        server = SimulatorServer(dump_dir=tmp_dir.name)
+        init_data, repo = server.new_test("unittest_basic_metadata_hash_support")
+        repo.compute_metafile_hashes_length = True
+        repo.update_snapshot()  # v2
+        self.assertEqual(
+            Metadata.from_bytes(repo.signed_mds[Snapshot.type][-1]).signed.version, 2
+        )
+
+        expected_hashes, expected_length = repo._compute_hashes_and_length(
+            Snapshot.type
+        )
+
+        self.assertEqual(
+            Metadata.from_bytes(
```

Comment:
You can add the missing annotation and silence mypy with

`Metadata[Timestamp].from_bytes(`

Reply:
https://github.com/theupdateframework/tuf-conformance/pull/188/commits/984b4abfb0ee3f797268031d19c97be29922493c